server/src/test/java/org/elasticsearch/rest/action/document/RestGetSourceActionTests.java

```diff
@@ -92,53 +93,34 @@ public void testRestGetSourceActionWithMissingDocumentSource() {
     /**
      * test deprecation is logged if type is used in path
      */
-    public void testTypeInGetPath() {
-        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
-            .withHeaders(Map.of("Accept", compatibleMediaType))
-            .withMethod(RestRequest.Method.HEAD)
-            .withPath("/some_index/some_type/id/_source")
-            .build();
-        dispatchRequest(request);
-        assertWarnings(RestGetSourceAction.TYPES_DEPRECATION_MESSAGE);
-    }
-
-    public void testTypeInHeadPath() {
-        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
-            .withHeaders(Map.of("Accept", compatibleMediaType))
-            .withMethod(RestRequest.Method.GET)
-            .withPath("/some_index/some_type/id/_source")
-            .build();
-        dispatchRequest(request);
-        assertWarnings(RestGetSourceAction.TYPES_DEPRECATION_MESSAGE);
+    public void testTypeInPath() {
+        for (RestRequest.Method method : Arrays.asList(RestRequest.Method.GET, RestRequest.Method.HEAD)) {
+            RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
+                .withHeaders(Map.of("Accept", compatibleMediaType))
+                .withMethod(method)
+                .withPath("/some_index/some_type/id/_source")
+                .build();
+            dispatchRequest(request);
+            assertWarnings(RestGetSourceAction.TYPES_DEPRECATION_MESSAGE);
+        }
     }
 
     /**
      * test deprecation is logged if type is used as parameter
      */
-    public void testTypeParameterAndGet() {
-        Map<String, String> params = new HashMap<>();
-        params.put("type", "some_type");
-        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
-            .withHeaders(Map.of("Accept", compatibleMediaType))
-            .withMethod(RestRequest.Method.GET)
-            .withPath("/some_index/_source/id")
-            .withParams(params)
-            .build();
-        dispatchRequest(request);
-        assertWarnings(RestGetSourceAction.TYPES_DEPRECATION_MESSAGE);
-    }
-
-    public void testTypeParameterAndHead() {
+    public void testTypeParameter() {
         Map<String, String> params = new HashMap<>();
         params.put("type", "some_type");
-        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
-            .withHeaders(Map.of("Accept", compatibleMediaType))
-            .withMethod(RestRequest.Method.HEAD)
-            .withPath("/some_index/_source/id")
-            .withParams(params)
-            .build();
-        dispatchRequest(request);
-        assertWarnings(RestGetSourceAction.TYPES_DEPRECATION_MESSAGE);
+        for (RestRequest.Method method : Arrays.asList(RestRequest.Method.GET, RestRequest.Method.HEAD)) {
```

Comment:
These test changes while nice aren't really related to the rest of the PR, maybe revert those to keep the change isolated?

Reply:
good idea, I will revert the test cases and will follow up in a separate PR